### DISCUSSION TITLE
Step-ca generated expired/invalid certs

### DISCUSSION THREAD

**Discussion by SpiderUnderUrBed** [smallstep/certificates#1871] [Q&A] [answered]
I am using step-ca with ansible, i automated the providers, starting of the server, certificate generation and the deploying of the nginx container here:
create-cert.yml:
https://pastebin.com/E9ZaWgcA
nginx.yml:
https://pastebin.com/SkJD8Zej

You dont need to understand ansible to see whats going on
Now the issue:
![image](https://github.com/smallstep/certificates/assets/71126384/8c197762-4c1b-4b33-a0d0-02db89bd257f)
Even if I try to attach the certificate one minuite after its creation it says its invalid, thats about all my issue. and i even set flags for its expirey date, idk why this is happening, and i would like help

my ca.json:
https://pastebin.com/X9E7RnC6

**Comment by tashian** [comment]
You may be dealing with a time difference between systems.
Try creating the certificate with `--not-before -5m` to give yourself some leeway on the Not Before value of the certificate.

**Comment by SpiderUnderUrBed** [comment]
I think the timing issue is sorted but i have another issue:
![image](https://github.com/smallstep/certificates/assets/71126384/3b0f93ae-2de1-4c47-bbcd-623ace6d9726)
Looking at the ansible configuration and the image, i am using the correct certificates right? How do i fix this?
My certs arnt password protected, also here is the error:
`Upload failed: Certificate is not valid (Could not read certificate from /tmp/2041df3c-ff12-490f-b382-256c59bdd55f/tmp 20C081857F000000:error:1608010C:STORE routines:ossl_store_handle_load_result:unsupported:../crypto/store/store_result.c:151: Unable to load certificate )`

**Comment by SpiderUnderUrBed** [comment]
Ah nevermind that issue, turns out I have been uploading my key twice. I still get the "certificate has expired issue" though.

**Comment by SpiderUnderUrBed** [comment, marked as answer]
Ah fixed, turns out, that first I used two keys accidentally, than adding my cert, and just now i realized i had a typo in the just before flag, it looked like -1d, it does not support days. and the '-' yeilded an error so it kept on using the outdated one, thanks for your help!
